src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -312,8 +385,23 @@ public static String getContainerSettingJson(int startTagIndex, int numToProcces
       endTagIndex = TagInfoManager.getTagInfoList().size();
     }
     for (int i = startTagIndex; i < endTagIndex; i++) {
+    
       if (TagInfoManager.getTagInfoList().get(i) != null) {
-        String tagName = ((TagInfo) TagInfoManager.getTagInfoList().get(i)).getName();
+        TagInfo currentTag = ((TagInfo) TagInfoManager.getTagInfoList()).get(i);
+        String tagName = currentTag.getName();
+
+        if (currentTag.getType() == TagType.FLOAT) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.INTEGER) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.BOOLEAN) {
+          typeID = "HMS-boolean-type-" + getFlexyName();
+        } else if (currentTag.getType() == TagType.DWORD) {
+          typeID = "HMS-number-type-" + getFlexyName();
+        } else {
+          // string type should work with everything. This is the best default.
```

Comment:
The flexy may add additional types, you are not guaranteed a string here.

Reply:
fair enough. An error message for not supported types makes more sense then